src/cryptosettings.cpp

```diff
@@ -228,13 +228,13 @@ bool CryptoSettings::readEncryptedChachaPolyFile(Version fileVersion,
 bool CryptoSettings::writeFile(QIODevice& device,
                                const QSettings::SettingsMap& map) {
   logger.debug() << "Writing the settings file";
-  CryptoSettings::Version version = NoEncryption;
 
-  if (s_instance) {
-    version = s_instance->getSupportedVersion();
+  // Without a SryptoSettings we can only write plaintext settings.
```

Comment:
When do we not have SyrptoSettings? Is it better to fail to plaintext, or fail to not saving the key?

Reply:
Oops, that was a bit of a typo :)

When `s_instance` is null, this means that there is no platform provider for fetching keys in which case the only format we can possibly support is the plaintext settings (which is what `writeJsonFile()` writes). Nothing has really changed in this method in terms of logic, I just reorganized it to eliminate the local variable.